src/main/java/ch/njol/skript/expressions/ExprEntitySound.java

```diff
@@ -0,0 +1,154 @@
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ExprEntitySound extends SimpleExpression<String> {
+
+	static {
+		if (Skript.methodExists(LivingEntity.class, "getDeathSound")) {
+			Skript.registerExpression(ExprEntitySound.class, String.class, ExpressionType.COMBINED,
+				"[the] (damage|hurt) sound[s] of %livingentities%",
+				"%livingentities%'[s] (damage|hurt) sound[s]",
+
+				"[the] death sound[s] of %livingentities%",
+				"%livingentities%'[s] death sound[s]",
+
+				"[the] [high:(tall|high)|(low|normal)] fall [damage] sound[s] [from [[a] height [of]] %-number%] of %livingentities%",
+				"%livingentities%'[s] [high:(tall|high)|low:(low|normal)] fall [damage] sound[s] [from [[a] height [of]] %-number%]",
+
+				"[the] swim[ming] sound[s] of %livingentities%",
+				"%livingentities%'[s] swim[ming] sound[s]",
+
+				"[the] [fast:(fast|speedy)] splash sound[s] of %livingentities%",
+				"%livingentities%'[s] [fast:(fast|speedy)] splash sound[s]",
+
+				"[the] eat[ing] sound of %livingentities% [(with|using|[while] eating [a]) %-itemtype%]",
+				"%livingentities%'[s] eat[ing] sound",
+
+				"[the] drink[ing] sound of %livingentities% [(with|using|[while] drinking [a]) %-itemtype%]",
+				"%livingentities%'[s] drink[ing] sound");
+		}
+	}
+
+	private static final int DAMAGE = 0;
+	private static final int DEATH = 2;
+	private static final int FALL = 4;
+	private static final int SWIM = 6;
+	private static final int SPLASH = 8;
+	private static final int EAT_WITH_ITEM = 10;
+	private static final int EAT = 11;
+	private static final int DRINK_WITH_ITEM = 12;
+	private static final int DRINK = 13;
+
+	private int sound;
+	private boolean big;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<Number> height;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<LivingEntity> entities;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<ItemType> item;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		sound = matchedPattern;
+		big = parseResult.hasTag("high") || parseResult.hasTag("fast");
+		if (matchedPattern == FALL || matchedPattern == FALL + 1)
+			height = (Expression<Number>) exprs[0];
+		if (matchedPattern == EAT_WITH_ITEM || matchedPattern == DRINK_WITH_ITEM)
+			item = (Expression<ItemType>) exprs[1];
+		entities = (Expression<LivingEntity>) ((matchedPattern == FALL) ? exprs[1] : exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected String @Nullable [] get(Event event) {
+		Set<String> sounds = new HashSet<>();
+		for (LivingEntity entity : entities.getArray(event)) {
+			Sound sound = getEntitySound(entity, event);
+			if (sound != null)
+				sounds.add(sound.name());
+		}
+		return sounds.toArray(String[]::new);
```

Comment:
Can use a stream here. Import required.
```suggestion
		return entities.stream(event)
				.map(entity -> getEntitySound(entity, event))
				.map(Objects::nonNull)
				.toArray(String[]::new);
```